src/test.js

```diff
@@ -550,6 +550,13 @@ function test( testName, expected, callback, async ) {
 	if ( arguments.length === 2 ) {
 		callback = expected;
 		expected = null;
+	} else if ( arguments.length === 3 && typeof expected !== "function" ) {
+		if ( global.console && global.console.warn ) {
+			global.console.warn(
+				"'expected' argument of QUnit.test is deprecated. " +
+				"Call the expect() function explicitly instead."
```

Comment:
> Does that sounds good to you?

It sounds great, but I would use a `QUnit.warnings` instead of `deprecationWarnings`, with that we can reuse it to any other warning type, even if it's not a deprecated one.

> Done. Added in core/utilities.js.

Did you pushed the commit with it? I would love to see it.